server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/LocalResources.java

```diff
@@ -0,0 +1,113 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailetcontainer;
+
+import java.util.EnumSet;
+import java.util.Locale;
+
+import javax.inject.Inject;
+import javax.mail.internet.ParseException;
+
+import org.apache.james.core.Domain;
+import org.apache.james.core.MailAddress;
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.domainlist.api.DomainListException;
+import org.apache.james.rrt.api.RecipientRewriteTable;
+import org.apache.james.rrt.api.RecipientRewriteTableException;
+import org.apache.james.rrt.lib.Mapping;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.api.UsersRepositoryException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.fge.lambdas.Throwing;
+
+public class LocalResources {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LocalResources.class);
+    private static final EnumSet<Mapping.Type> ALIAS_TYPES = EnumSet.of(Mapping.Type.Alias, Mapping.Type.DomainAlias);
+
+    private final UsersRepository localusers;
+    private final DomainList domains;
+    private final RecipientRewriteTable recipientRewriteTable;
+
+    @Inject
+    public LocalResources(UsersRepository localusers, DomainList domains, RecipientRewriteTable recipientRewriteTable) {
+        this.localusers = localusers;
+        this.domains = domains;
+        this.recipientRewriteTable = recipientRewriteTable;
+    }
+
+    public boolean isLocalServer(Domain domain) {
+        try {
+            return domains.containsDomain(domain);
+        } catch (DomainListException e) {
+            LOGGER.error("Unable to retrieve domains", e);
+            return false;
```

Comment:
this is a bad fallback, not being able to do a lookup should not end in considering a domain to not be handle by the server.